backend/src/main/java/io/diveni/backend/service/news/GithubApiService.java

```diff
@@ -0,0 +1,88 @@
+package io.diveni.backend.service.news;
+
+import io.diveni.backend.controller.ErrorMessages;
+import io.diveni.backend.model.news.PullRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+import static io.diveni.backend.controller.NewsController.*;
+
+@Service
+public class GithubApiService {
+
+  private static final List<String> VALID_STATES = List.of("open", "closed", "all");
+  private static final int MAX_PER_PAGE = 100;
+
+  private static final String API_VERSION_HEADER = "X-GitHub-Api-Version";
+
+  @Value("${vsc.github.api.pr.get}")
+  private String url;
+
+  @Value("${vsc.github.access-token}")
+  private String authToken;
+
+  @Value("${vsc.github.api.version}")
+  private String apiVersion;
+
+  private static final Logger logger = LoggerFactory.getLogger(GithubApiService.class);
+  private RestTemplate client;
+
+  @Autowired
+  public GithubApiService(RestTemplate client) {
+    this.client = client;
+  }
+
+  public PullRequest[] getPullRequests(String state, String sort, String direction, boolean isMerged, int perPage, int page) {
```

Comment:
I'm not sure. Do we even need sort, direction & isMerged as parameters here? If I didnt miss something, these values are hard coded in the frontend and there is no button to change them.

Reply:
The reason for adding those params is extensibility in the future, if needed. For example lets say in the future there is new feature request - add options which give the user ability to sort the prs in the 'What's new page' by different params. This way only the frontend will need changes